### ISSUE TITLE
Pre-allocate objects instead of creating new ones

### ISSUE THREAD

**Issue by tpoisot** [MichielStock/Just_for_fun#2]
In some functions, you could modify an object created elsewhere instead of creating and returning a new one. This would save some memory

https://github.com/MichielStock/Just_for_fun/blob/189e0d050dd94f0fdf9c8a3d50da02235bf99c3f/StatisticalPhysics/statisticalphysics.jl#L23

See this part of the manual: https://docs.julialang.org/en/v0.6.1/manual/performance-tips/#Pre-allocating-outputs-1

**Comment by tpoisot** [comment]
In `generate_circles`, you could allocate `circles` at the beginning, and modify the circle generation function to write in this object.

The `Traceur` package is helpful to catch possible performance issues: https://github.com/MikeInnes/Traceur.jl

**Comment by MichielStock** [comment]
Indeed, did this only in `simulate_circles`. The reason why I did not do this is because `generate_circles` is less general. I guess the best is to make a function `fill_circles!` that does that.
